app/controllers/propertyLinking/AreYouSureController.scala

```diff
@@ -65,21 +65,42 @@ class AreYouSureController @Inject()(
 
   def submitRatingListYears(chosenListYear: String): Action[AnyContent] = authenticated.async { implicit request =>
     if (featureSwitch.isAgentListYearsEnabled) {
-      manageAgentSessionRepository.get[AgentSummary].map {
+      manageAgentSessionRepository.get[AgentSummary].flatMap {
```

Comment:
Because this code is mostly reused across this and the AreYouSureMultipleController, would it be better moving into a service class?